class.jetpack-cli.php

```diff
@@ -997,6 +998,11 @@ public function sync( $args, $assoc_args ) {
 						} else {
 							WP_CLI::log( __( 'Sent more data to WordPress.com', 'jetpack' ) );
 						}
+
+						// Immediate Full Sync does not wait for WP.com to process data so we need to enforce a wait.
+						if ( false !== strpos( get_class( Modules::get_module( 'full-sync' ) ), 'Full_Sync_Immediately' ) ) {
+							sleep( 15 );
```

Comment:
Do we want a hardcoded sleep here? I'm not sure how good it would be to force a 15 second sleep if Max Exec on some hosts is lower than that.

Reply:
Yes, there needs to be an enforced sleep for Full Sync. The current logic doesn't wait for WP.com to process the data. Without the sleep it will continually push data to WP.com and lead to concurrent request errors.

Reply:
Ran more testing on this and confirmed that we do want a sleep on this. Full Sync doesn't queue data so each iteration does a query on the db. Without the wait the cli could cause concerns to the site's database and degrade overall performance.